src/backend/distributed/commands/alter_table.c

```diff
@@ -1517,6 +1524,33 @@ CreateWorkerChangeSequenceDependencyCommand(char *sequenceSchemaName, char *sequ
 }
 
 
+/*
+ * CreateUsingAccessMethodClauseForMatViewIfExists generates and returns "USING ..." string
+ * if there's an access method set to the view with the given oid.
+ */
+static char *
+CreateUsingAccessMethodClauseForMatViewIfExists(Oid viewOid)
+{
+	StringInfo usingClause = makeStringInfo();
+	Relation relation = try_relation_open(viewOid, AccessShareLock);
+	if (relation == NULL)
+	{
+		ereport(ERROR, (errmsg("cannot complete operation "
+							   "because no such view exists")));
+	}
+
+	Oid accessMethodOid = relation->rd_rel->relam;
+	if (OidIsValid(accessMethodOid))
+	{
+		char *accessMethodName = get_am_name(accessMethodOid);
```

Comment:
I'm not sure if it is certainly necessary but I think we can check if we actually received a result from this function.